components/ws-manager/cmd/run.go

```diff
@@ -133,13 +136,32 @@ var runCmd = &cobra.Command{
 				grpc_opentracing.UnaryServerInterceptor(grpc_opentracing.WithTracer(opentracing.GlobalTracer())),
 			)),
 		}
-		if cfg.RPCServer.TLS.Certificate != "" && cfg.RPCServer.TLS.PrivateKey != "" {
-			creds, err := credentials.NewServerTLSFromFile(cfg.RPCServer.TLS.Certificate, cfg.RPCServer.TLS.PrivateKey)
+		if cfg.RPCServer.TLS.CA != "" && cfg.RPCServer.TLS.Certificate != "" && cfg.RPCServer.TLS.PrivateKey != "" {
+			cert, err := tls.LoadX509KeyPair(cfg.RPCServer.TLS.Certificate, cfg.RPCServer.TLS.PrivateKey)
 			if err != nil {
 				log.WithError(err).WithField("crt", cfg.RPCServer.TLS.Certificate).WithField("key", cfg.RPCServer.TLS.PrivateKey).Fatal("could not load TLS keys")
 			}
+			certPool := x509.NewCertPool()
+			b, err := ioutil.ReadFile(cfg.RPCServer.TLS.CA)
+			if err != nil {
+				log.WithError(err).WithField("ca", cfg.RPCServer.TLS.CA).Fatal("could not load CA")
+			}
+			if !certPool.AppendCertsFromPEM(b) {
+				log.WithError(err).WithField("ca", cfg.RPCServer.TLS.CA).Fatal("failed to append CA")
+			}
+
+			creds := credentials.NewTLS(&tls.Config{
+				ClientAuth:   tls.RequireAndVerifyClientCert,
```

Comment:
This reads like client authentication, which is also nice, but involves more certifactes/keys to be shoveled around. I hoped could get away with a `token`-based client auth for now.
Server authentication is a tad simpler and explained: https://grpc.io/docs/guides/auth/#with-server-authentication-ssltls

Reply:
See my comment here: https://github.com/gitpod-io/gitpod/pull/3658#discussion_r603903638

I thought we would like to go with mutual TLS. From the design doc:

> mutual trust: we need to ensure that the ws-manager that's trying to register is one of ours, and not some impersonator. At the same time, our ws-manager must be sure that the ws-manager-bridge they're talking to is one of ours.
> We would use mutual TLS (mTLS) on the registry API interface of ws-manager-bridge to ensure mutual trust. gRPC supports mTLS.